src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -114,14 +143,106 @@
 	public void init() throws ServletException {
 		M_log.debug(" Servlet init(): Called");
 		appExtPropertiesFile = Utils.loadProperties(CCM_PROPERTY_FILE_PATH);
-		appExtSecurePropertiesFile = Utils.loadProperties(CCM_SECURE_PROPERTY_FILE_PATH);		
+		appExtSecurePropertiesFile = Utils.loadProperties(CCM_SECURE_PROPERTY_FILE_PATH);
+		oacf = new OauthCredentialsFactory(appExtSecurePropertiesFile);
 	}
 
 	public void fillContext(Context context, HttpServletRequest request) {
-		M_log.debug("fillContext() called");		
-		ViewToolContext vtc = (ViewToolContext)context;
-		//test code
-		//vtc.put("variable", "happiness");
+		M_log.debug("fillContext() called");
+
+		if ( BASIC_LTI_LAUNCH_REQUEST.equals(request.getParameter(LTI_MESSAGE_TYPE))) {
+			Map<String, String> ltiValues = new HashMap<String, String>();
```

Comment:
The else clause should be a separate method, not lines and lines of nested code.